### DISCUSSION TITLE
Run multiple queries dynamically

### DISCUSSION THREAD

**Discussion by Annie-khalid** [supabase/supabase#884] [General]
# Feature request
I want to run multiple queries on my data , my tables have foreign relations. Multiple queries are working on single table but not on foreign table so what is the best way to run conditions on tables (including foreign table) at a time.
## Is your feature request related to a problem? Please describe

A clear and concise description of what you want and what your use case is.

**Comment by kiwicopple** [comment]
HI @Annie-khalid 

>  is the best way to run conditions on tables (including foreign table) at a time

Is this to fetch the data? Like this? https://supabase.io/docs/client/select#query-foreign-tables

What are the queries that you are trying to run? If you have some code snippets available that would be great 👍

**Comment by Annie-khalid** [comment]
Yes i want to fetch data but if the condition is satisfied. For example i want to get all data  where source( that is column in cv table) is "website".
**So query will be like this**
property = "cv"
condition = "eq"
value = "website"

**My code for filtering** 
{

			// define query
			let q = supabase.from('profile').select("*,cv(*),interview(*),education(*),projects(*),experience(*)")

			// Loop over the queries
			for (var index of Object.keys(this.state.queries)) {

				// Get query
				var query = this.state.queries[index];

				// Normalize the array
				var value = query.condition === "cs" ? `{${query.value.toString()}}` : query.value;

				// apply filter
				q.filter(query.property, query.condition, value)

			}

			// get data 
			var {data, error} = await q
		
		} 


**Note:** i want to check conditions on foreign table's attributes as well.

**Comment by kiwicopple** [comment]
```
property = "cv"
condition = "eq"
value = "website"
```

it's a bit hard to know without more details, but I suspect that you will want to put the "CV" table at the root of your query?

```js
supabase
  .from('cv')
  .select(`
    *, profiles(
      *, interview(*),education(*),projects(*),experience(*)
    )
  `)
  .eq('website', SOME_WEBSITE)
```

I'm going to convert this one to a discussion and if you can provide any more feedback hopefully everyone can give you some more ideas
